build-docker/Dockerfile-factory

```diff
@@ -0,0 +1,15 @@
+FROM espressif/idf:v5.3.1
```

Comment:
We probably want to use arguments for dependency versioning. Also, we are currently using idf v5.3.2 and moving to 5.4.0 soon.